packages/loaders/src/Progress.spec.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React from 'react';
+import { render } from 'garden-test-utils';
+import Progress from './Progress';
+const { zdColorBlue600 } = require('@zendeskgarden/css-variables');
+
+describe('Progress', () => {
+  describe('without a value', () => {
+    it('renders a progress bar with 0% progress', () => {
+      const { getByRole } = render(<Progress />);
+
+      expect(getByRole('progressbar')).toHaveAttribute('aria-valuenow', '0');
+    });
+  });
+
+  it('renders a progress bar with correct aria attributes', () => {
+    const { getByRole } = render(<Progress value={40} />);
+
+    const progress = getByRole('progressbar');
+
+    expect(progress).toHaveAttribute('role', 'progressbar');
+    expect(progress).toHaveAttribute('aria-valuemin', '0');
+    expect(progress).toHaveAttribute('aria-valuenow', '40');
+    expect(progress).toHaveAttribute('aria-valuemax', '100');
+  });
+
+  it('renders a progress indicator', () => {
+    const { getByRole } = render(<Progress value={40} />);
+
+    expect(getByRole('progressbar').firstChild).toHaveStyleRule('width', '40%');
+  });
+
+  describe('when given a value below zero', () => {
+    it('renders a progress bar with 0% progress', () => {
+      const { getByRole } = render(<Progress value={-42} />);
+
+      expect(getByRole('progressbar')).toHaveAttribute('aria-valuenow', '0');
+    });
+  });
+
+  describe('when given a value above 100', () => {
+    it('renders a progress bar with 100% progress', () => {
+      const { getByRole } = render(<Progress value={666} />);
+
+      expect(getByRole('progressbar')).toHaveAttribute('aria-valuenow', '100');
+    });
+  });
+
+  describe('with a small size', () => {
+    it('renders a small progress bar', () => {
+      const { getByRole } = render(<Progress value={42} size="small" />);
+
+      expect(getByRole('progressbar').firstChild).toHaveStyleRule('height', '2px');
```

Comment:
Not requesting a change, but would like to put my thoughts out there for discussion.

I feel like looking up a target element via `.firstChild` _can_ be bit brittle. This is probably more true for application code than library code, but it's always stuck with me. For example, by adding a non-content styled `div`, these tests would break:

```jsx
const Progress = ({ value, size, ...other }) => {
  const percentage = Math.max(0, Math.min(100, value));

  return (
    <ProgressBackground
      data-garden-id={COMPONENT_ID}
      data-garden-version={PACKAGE_VERSION}
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow={percentage}
      role="progressbar"
      size={size}
      {...other}
    >
      <div id="tracking-pixel"></div>
      <ProgressIndicator value={percentage} size={size} />
    </ProgressBackground>
  );
};
```

I know this is a pattern Garden has had (and I'm new!) so I'm **not** suggesting we change this. Also, this is probably not that important in this scenario because the code example above is contrived and would probably never happen in Garden. But just wanted to put my thoughts out on targeting elements by using `.firstChild`.

Reply:
We _could_ target this with a `data-garden-test-id`, which is removed during the `production` builds, but not sure it would be worth it in this use-case. Here is the logic for that https://github.com/zendeskgarden/react-components/blob/master/babel.config.js#L24-L28

Reply:
I don't think it would be worth it for this case.